[JfrEventSubstitution] HotSpotResolvedJavaMethodImpl.getMetaspaceMethod() not available in plain OpenJDK 11.0.11

This is latest master at revision: c656cf921a39eb73f96ff1edef12b85eaeeb5b7f

See https://github.com/graalvm/mandrel/issues/262 for the original issue. Building HelloWorld native-image with `-H+AllowVMInspection` fails with:
```
$ native-image -H:+AllowVMInspection HelloWorld                            
[helloworld:337555]    classlist:   1,193.54 ms,  0.96 GB
[helloworld:337555]        (cap):     447.76 ms,  0.96 GB
[helloworld:337555]        setup:   1,205.43 ms,  0.96 GB
Fatal error:com.oracle.svm.core.util.VMError$HostedError: java.lang.NoSuchMethodException: jdk.vm.ci.hotspot.HotSpotResolvedJavaMethodImpl.getMetaspaceMethod()
	at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:72)
	at com.oracle.svm.jfr.JfrEventSubstitution.setPublicModifier(JfrEventSubstitution.java:123)
	at com.oracle.svm.jfr.JfrEventSubstitution.changeWriterResetMethod(JfrEventSubstitution.java:98)
	at com.oracle.svm.jfr.JfrEventSubstitution.<init>(JfrEventSubstitution.java:63)
	at com.oracle.svm.jfr.JfrFeature.duringSetup(JfrFeature.java:138)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$setupNativeImage$18(NativeImageGenerator.java:928)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
	at com.oracle.svm.hosted.NativeImageGenerator.setupNativeImage(NativeImageGenerator.java:928)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:571)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$2(NativeImageGenerator.java:497)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoSuchMethodException: jdk.vm.ci.hotspot.HotSpotResolvedJavaMethodImpl.getMetaspaceMethod()
	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2475)
	at com.oracle.svm.jfr.JfrEventSubstitution.setPublicModifier(JfrEventSubstitution.java:106)
	... 14 more
[helloworld:337555]      [total]:   2,443.49 ms,  0.96 GB
# Printing build artifacts to: /home/zakkak/code/mandrel-packaging/helloworld.build_artifacts.txt
Error: Image build request failed with exit status 1
```

`getMetaspaceMethod()` which is reflectively accessed in JfrEventSubstitution, got introduced in OpenJDK 13 with JDK-8220623. This bug has not been backported to OpenJDK 11u which now breaks Mandrel-builds which want to use that flag. A simple fallback to `getMetaspacePointer()` should suffice to also support those builds.

--- a/substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrEventSubstitution.java
+++ b/substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrEventSubstitution.java
@@ -103,9 +103,10 @@ public class JfrEventSubstitution extends SubstitutionProcessor {
     private static void setPublicModifier(ResolvedJavaMethod m) {
         try {
             Class<?> method = m.getClass();
-            Method metaspaceMethodM = method.getDeclaredMethod("getMetaspaceMethod");
+            Method metaspaceMethodM = method.getDeclaredMethod("getMetaspacePointer");
             metaspaceMethodM.setAccessible(true);
             long metaspaceMethod = (Long) metaspaceMethodM.invoke(m);
+            VMError.guarantee(metaspaceMethod != 0);
             // Checkstyle: stop
             Class<?> hotSpotVMConfigC = Class.forName("jdk.vm.ci.hotspot.HotSpotVMConfig");
             // Checkstyle: resume

